### ISSUE TITLE
Imported module and module that imports this module cannot have the same name

### ISSUE THREAD

**Issue by mikeconley12** [nestjs/nest#3512]
### Input code

**a.module.ts**
```
import { Controller, Module } from "@nestjs/common";
import { AProvider } from "./third-party-a.module";
import { AModule as ThirdPartyAModule } from "./third-party-a.module";

@Controller("example")
class AController {
  constructor(private readonly aProvider: AProvider) {
  }
}

@Module({
  imports: [ThirdPartyAModule],
  controllers: [AController]
})
export class AModule {
}
```

**third-party-a.module.ts**

```
import { Injectable, Module } from "@nestjs/common";

@Injectable()
export class AProvider {
}

@Module({
  providers: [AProvider],
  exports: [AProvider]
})
export class AModule {
}

```

**app.module.ts**

```
import { Module } from "@nestjs/common";
import { AModule } from "./a.module";

@Module({
  imports: [
    AModule
  ]
})
export class AppModule {
}
```

### Current behavior

I have the following error: 

```
[Nest] 14296   - 11/30/2019, 5:45:20 PM   [NestFactory] Starting Nest application...
[Nest] 14296   - 11/30/2019, 5:45:20 PM   [InstanceLoader] AppModule dependencies initialized +11ms
[Nest] 14296   - 11/30/2019, 5:45:20 PM   [ExceptionHandler] Nest can't resolve dependencies of the AController (?). Please make sure that the argument AProvider at index [0] is available in the AModule context.

Potential solutions:
- If AProvider is a provider, is it part of the current AModule?
- If AProvider is exported from a separate @Module, is that module imported within AModule?
  @Module({
    imports: [ /* the Module containing AProvider */ ]
  })
 +1ms
Error: Nest can't resolve dependencies of the AController (?). Please make sure that the argument AProvider at index [0] is available in the AModule context.

Potential solutions:
- If AProvider is a provider, is it part of the current AModule?
- If AProvider is exported from a separate @Module, is that module imported within AModule?
  @Module({
    imports: [ /* the Module containing AProvider */ ]
  })

    at Injector.lookupComponentInExports (/home/username/project/node_modules/@nestjs/core/injector/injector.js:189:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)

```

If I rename one of the modules, the error will disappear.
For example, I can rename `AModule` from **a.module.ts**

```
@Module({
  imports: [ThirdPartyAModule],
  controllers: [AController]
})
export class AModule2 {
}
```

```
@Module({
  imports: [
    AModule2
  ]
})
export class AppModule {
}
```

In this case, I have no error: 

```
[Nest] 14526   - 11/30/2019, 5:49:42 PM   [NestFactory] Starting Nest application...
[Nest] 14526   - 11/30/2019, 5:49:42 PM   [InstanceLoader] AppModule dependencies initialized +12ms
[Nest] 14526   - 11/30/2019, 5:49:42 PM   [InstanceLoader] AModule dependencies initialized +1ms
[Nest] 14526   - 11/30/2019, 5:49:42 PM   [InstanceLoader] AModule2 dependencies initialized +1ms
[Nest] 14526   - 11/30/2019, 5:49:42 PM   [RoutesResolver] AController {/example}: +7ms
[Nest] 14526   - 11/30/2019, 5:49:42 PM   [NestApplication] Nest application successfully started +3ms
```

### Expected behavior

I am not sure if this is a bug, but in my case it would be useful to have the same name for the module and the imported module.

### Environment

```
"@nestjs/common": "6.10.1",
"@nestjs/core": "6.10.1",

- Node version: v10.17.0
- Platform: Linux
```

**Comment by kamilmysliwiec** [comment]
This should be fixed in `6.10.6`. Please, let me know if you're still facing this issue.

**Comment by mikeconley12** [comment]
@kamilmysliwiec thank you!
